[Bug]: Invalid hash links in how-to-open-a-pull-request and moderator-handbook docs

### Describe the bug

During the local build process for the documentation site, the Starlight link validator catches invalid hash links pointing to non-existent headings in the markdown files. Specifically, it flags invalid hash errors for links inside how-to-open-a-pull-request.mdx and moderator-handbook.mdx.

### To Reproduce

1. Clone the freeCodeCamp repository and set up the docs site locally.
2. Run the build/dev command that triggers Starlight's search indexing and link validation.
3. Check the terminal output.
4. See the [ERROR] ✗ Found X invalid links... pointing to the invalid hashes.

### Expected behavior

The build process should complete without invalid hash errors, and all internal documentation links should point to correctly formatted and existing markdown headings.

### Screenshots

17:43:10 [ERROR] ✗ Found 8 invalid links in 7 files.
...
17:43:10 ▶ how-to-open-a-pull-request/
17:43:10   └─ /how-to-setup-freecodecamp-locally/#making-changes-locally - invalid hash
...
17:43:10 ▶ moderator-handbook/
17:43:10   └─ /faq/#additional-assistance - invalid hash

### Desktop OS

Fedora 43

### Desktop Browser

Google Chrome

### Desktop Browser Version

_No response_

### Mobile Device

_No response_

### Mobile OS

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional context

I have already investigated this and prepared a fix for these two specific hash links. I will be opening a Pull Request shortly with the following changes:

In how-to-open-a-pull-request.mdx, updated the link from /how-to-setup-freecodecamp-locally/#making-changes-locally to /how-to-setup-freecodecamp-locally/#set-up-syncing-from-parent, which matches the actual heading.

In moderator-handbook.mdx, replaced /faq/#additional-assistance with /faq/#where-can-i-get-help-if-im-stuck-on-something-not-covered-here, matching the real FAQ section heading.

--- a/src/content/docs/how-to-open-a-pull-request.mdx
+++ b/src/content/docs/how-to-open-a-pull-request.mdx
@@ -127,5 +127,5 @@ Our moderators will now take a look and leave you feedback. Please be patient wi
 And as always, feel free to ask questions on the ['Contributors' category on our forum](https://forum.freecodecamp.org/c/contributors) or [the contributors chat room](https://discord.gg/PRyKn3Vbay).
 
 :::tip
-If you are to be contributing more pull requests, we recommend you read the [making changes and syncing](/how-to-setup-freecodecamp-locally/#set-up-syncing-from-parent) guidelines to avoid having to delete your fork.
+If you are to be contributing more pull requests, we recommend you read the [making changes and syncing](/how-to-contribute-to-the-codebase/#contributing-to-the-codebase) guidelines to avoid having to delete your fork.
 :::

--- a/src/content/docs/how-to-open-a-pull-request.mdx
+++ b/src/content/docs/how-to-open-a-pull-request.mdx
@@ -127,5 +127,5 @@ Our moderators will now take a look and leave you feedback. Please be patient wi
 And as always, feel free to ask questions on the ['Contributors' category on our forum](https://forum.freecodecamp.org/c/contributors) or [the contributors chat room](https://discord.gg/PRyKn3Vbay).
 
 :::tip
-If you are to be contributing more pull requests, we recommend you read the [making changes and syncing](/how-to-setup-freecodecamp-locally/#making-changes-locally) guidelines to avoid having to delete your fork.
+If you are to be contributing more pull requests, we recommend you read the [making changes and syncing](/how-to-setup-freecodecamp-locally/#set-up-syncing-from-parent) guidelines to avoid having to delete your fork.
 :::

--- a/src/content/docs/moderator-handbook.mdx
+++ b/src/content/docs/moderator-handbook.mdx
@@ -27,7 +27,7 @@ freeCodeCamp is an inclusive community, and we need to keep it that way.
 We have a single [Code of Conduct](https://code-of-conduct.freecodecamp.org) that governs our entire community. The fewer the rules, the easier they are to remember. You can read those rules and commit them to memory [here](https://code-of-conduct.freecodecamp.org).
 
 :::note
-As a moderator, we would add you to one or more teams on GitHub, our community forums & chat servers. If you are missing access to a platform that you would like to moderate, please [reach out to a staff member](/faq/#additional-assistance).
+As a moderator, we would add you to one or more teams on GitHub, our community forums & chat servers. If you are missing access to a platform that you would like to moderate, please [reach out to a staff member](/faq/#where-can-i-get-help-if-im-stuck-on-something-not-covered-here).
 :::
 
 ## Moderating GitHub

